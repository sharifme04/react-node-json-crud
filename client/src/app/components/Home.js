import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import TableRow from "./TableRow";
import SearchField from "./SearchField";

const Home = ({ todoList, value, handleChange, isLoading }) => {
  const filteredTodoList = todoList
    ?.filter(
      (k) => k?.title?.toLowerCase().indexOf(value?.toLowerCase()) !== -1
    )
    .map((e) => <TableRow key={e.id} e={e} />);

  if (isLoading) {
    return (
      <div className="d-flex justify-content-center">
        <div className="spinner-border" role="status">
          <span className="visually-hidden">Loading...</span>
        </div>
      </div>
    );
  }

  return (
    <div>
      <div className="input-group d-flex justify-content-between">
        <Link to={`/todo/addnew`}>
          <button type="button" className="btn btn-primary">
            Add New
          </button>
        </Link>
        <SearchField handleChange={handleChange} value={value} />
        <div></div>
      </div>
      <hr></hr>
      <table className="table table-striped">
        <thead>
          <tr>
            <th scope="col">ID</th>
            <th scope="col">Title</th>
            <th scope="col">Description</th>
            <th scope="col">Date</th>
            <th scope="col">Edit</th>
            <th scope="col">Delete</th>
          </tr>
        </thead>
        <tbody>{filteredTodoList}</tbody>
      </table>
    </div>
  );
};

Home.propTypes = {
  todoList: PropTypes.array,
  value: PropTypes.any.isRequired,
  handleChange: PropTypes.func.isRequired,
  isLoading: PropTypes.bool
};

export default Home;
