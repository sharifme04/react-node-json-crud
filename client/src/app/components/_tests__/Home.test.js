import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import Home from "../Home";

const data = [
  {
    title: "some company name",
    description: "there is a description of the company",
    id: 767658,
    date: "18-08-2022",
  },
  {
    title: "sharif ",
    description: "This is a sharif text fosdmg, gmrat77777",
    date: "17-08-2022",
    id: 842223,
  },
  {
    title: "sharif 954t5",
    description: "This is a jkn5lkt text fosdmg, gmrat",
    id: 8422553,
    date: "17-08-2022",
  },
];

test("snapshot renders without props", () => {
  const component = renderer.create(
    <MemoryRouter>
      <Home />
    </MemoryRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot renders with props without data", () => {
  const component = renderer.create(
    <MemoryRouter>
      <Home
        todoList={[]}
        isLoading={true}
        handleChange={() => {}}
        value={"title"}
      />
    </MemoryRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot renders with props and data", () => {
  const component = renderer.create(
    <MemoryRouter>
      <Home
        todoList={data}
        isLoading={false}
        handleChange={() => {}}
        value={"title"}
      />
    </MemoryRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
