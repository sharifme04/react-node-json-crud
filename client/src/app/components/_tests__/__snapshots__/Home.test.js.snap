// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders with props and data 1`] = `
<div>
  <div
    className="input-group d-flex justify-content-between"
  >
    <a
      href="/todo/addnew"
      onClick={[Function]}
    >
      <button
        className="btn btn-primary"
        type="button"
      >
        Add New
      </button>
    </a>
    <div
      className="search-field"
    >
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="search by title"
        type="text"
        value="title"
      />
    </div>
    <div />
  </div>
  <hr />
  <table
    className="table table-striped"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          ID
        </th>
        <th
          scope="col"
        >
          Title
        </th>
        <th
          scope="col"
        >
          Description
        </th>
        <th
          scope="col"
        >
          Date
        </th>
        <th
          scope="col"
        >
          Edit
        </th>
        <th
          scope="col"
        >
          Delete
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`snapshot renders with props without data 1`] = `
<div
  className="d-flex justify-content-center"
>
  <div
    className="spinner-border"
    role="status"
  >
    <span
      className="visually-hidden"
    >
      Loading...
    </span>
  </div>
</div>
`;

exports[`snapshot renders without props 1`] = `
<div>
  <div
    className="input-group d-flex justify-content-between"
  >
    <a
      href="/todo/addnew"
      onClick={[Function]}
    >
      <button
        className="btn btn-primary"
        type="button"
      >
        Add New
      </button>
    </a>
    <div
      className="search-field"
    >
      <input
        className="form-control"
        placeholder="search by title"
        type="text"
      />
    </div>
    <div />
  </div>
  <hr />
  <table
    className="table table-striped"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          ID
        </th>
        <th
          scope="col"
        >
          Title
        </th>
        <th
          scope="col"
        >
          Description
        </th>
        <th
          scope="col"
        >
          Date
        </th>
        <th
          scope="col"
        >
          Edit
        </th>
        <th
          scope="col"
        >
          Delete
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
